soh/soh/Enhancements/kaleido.cpp

```diff
@@ -0,0 +1,443 @@
+            textColor = { 0x98, 0xFF, 0x44, 255 };
+        }
+
+        Matrix_Translate(mX, mY, 0.0f, MTXMODE_APPLY);
+//        Matrix_Scale(0.75f, 0.75f, 0.75f, MTXMODE_APPLY);
+
+        mEntryDl->push_back(gsSPMatrix(Matrix_NewMtx(play->state.gfxCtx, (char*)__FILE__, __LINE__), G_MTX_PUSH | G_MTX_LOAD | G_MTX_MODELVIEW));
+
+        // icon
+        if (!mAchieved) {
+            mEntryDl->push_back(gsDPSetGrayscaleColor(109, 109, 109, 255));
+            mEntryDl->push_back(gsSPGrayscale(true));
+        }
+        mEntryDl->push_back(gsDPSetPrimColor(0, 0, mIconColor.r, mIconColor.g, mIconColor.b, mIconColor.a));
+        mEntryDl->push_back(gsSPVertex(vtx, 4, 0));
+        LoadIconTex(mEntryDl);
+        mEntryDl->push_back(gsSP1Quadrangle(0, 2, 3, 1, 0));
+        mEntryDl->push_back(gsSPGrayscale(false));
+
+        // text
+        mEntryDl->push_back(gsDPSetPrimColor(0, 0, textColor.r, textColor.g, textColor.b, textColor.a));
+        for (size_t i = 0, vtxGroup = 0; i < numChar; i++) {
+            uint16_t texIndex = mText[i] - 32;
+
+            // A maximum of 64 Vtx can be loaded at once by gSPVertex, or basically 16 characters
+            // handle loading groups of 16 chars at a time until there are no more left to load.
+            // By this point 4 vertices have already been loaded for the preceding icon.
+            if (i % 16 == 0) {
+                size_t numVtxToLoad = std::min<size_t>(numChar - i, 16) * 4;
+                mEntryDl->push_back(gsSPVertex(&vtx[4 + (vtxGroup * 16 * 4)], numVtxToLoad, 0));
+                vtxGroup++;
+            }
+
+            if (texIndex != 0) {
+                auto texture = reinterpret_cast<uintptr_t>(Font_FetchCharTexture(texIndex));
+                auto vertexStart = static_cast<int16_t>(4 * (i % 16));
+
+                Gfx charTexture[] = {gsDPLoadTextureBlock_4b(texture, G_IM_FMT_I, FONT_CHAR_TEX_WIDTH,
+                                                             FONT_CHAR_TEX_HEIGHT, 0, G_TX_NOMIRROR | G_TX_CLAMP,
+                                                             G_TX_NOMIRROR | G_TX_CLAMP,
+                                                             G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD, G_TX_NOLOD)};
+                mEntryDl->insert(mEntryDl->end(), std::begin(charTexture), std::end(charTexture));
+                mEntryDl->push_back(
+                        gsSP1Quadrangle(vertexStart, vertexStart + 2, vertexStart + 3, vertexStart + 1, 0));
+            }
+        }
+        mEntryDl->push_back(gsSPPopMatrix(G_MTX_MODELVIEW));
+    }
+
+    Kaleido::Kaleido() {
+        const auto ctx = Rando::Context::GetInstance();
+        int yOffset = 2;
+        mEntries.push_back(std::make_shared<KaleidoEntryIconFlag>(gRupeeCounterIconTex, G_IM_FMT_IA, G_IM_SIZ_8b, 16, 16,
+                              Color_RGBA8{ 0xC8, 0xFF, 0x64, 255 }, FlagType::FLAG_RANDOMIZER_INF,
+                              static_cast<int>(RAND_INF_GREG_FOUND), 0, yOffset, "Greg"));
```

Comment:
Greg doesn't account for the green rupee color cosmetic.

Reply:
I thought I remembered us deciding at some point that it shouldn't.

Reply:
Yeah, I think Greg at least should remain green. It's kind of intrinsic to the character of the item.